components/brave_origin/brave_origin_policy_manager_unittest.cc

```diff
@@ -0,0 +1,288 @@
+#include "brave/components/brave_origin/brave_origin_policy_info.h"
+#include "brave/components/brave_origin/brave_origin_utils.h"
+#include "brave/components/brave_origin/pref_names.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_origin {
+
+class BraveOriginPolicyManagerTest : public testing::Test {
+ public:
+  BraveOriginPolicyManagerTest() = default;
+  ~BraveOriginPolicyManagerTest() override = default;
+
+  void SetUp() override {
+    pref_service_ = std::make_unique<TestingPrefServiceSimple>();
+    pref_service_->registry()->RegisterDictionaryPref(kBraveOriginPolicies);
+  }
+
+  void TearDown() override {
+    // Reset the singleton for clean test state
+    if (manager_) {
+      manager_->initialized_ = false;
+      manager_->browser_policy_definitions_.clear();
+      manager_->profile_policy_definitions_.clear();
+      manager_->local_state_ = nullptr;
+      manager_->observers_.Clear();
+      manager_ = nullptr;
+    }
+  }
+
+ protected:
+  void CreateTestPolicy(BraveOriginPolicyMap& policies,
+                        const std::string& pref_name,
+                        bool default_value,
+                        BraveOriginPolicyScope scope) {
+    policies.emplace(
+        pref_name, BraveOriginPolicyInfo(
+                       pref_name, default_value, scope, true,
+                       base::StrCat({"Test", pref_name, "Policy"}), pref_name));
+  }
+
+  BraveOriginPolicyMap CreateBrowserTestPolicies() {
+    BraveOriginPolicyMap test_policies;
+    CreateTestPolicy(test_policies, "test.global.pref", false,
+                     BraveOriginPolicyScope::kGlobal);
+    return test_policies;
+  }
+
+  BraveOriginPolicyMap CreateProfileTestPolicies() {
+    BraveOriginPolicyMap test_policies;
+    CreateTestPolicy(test_policies, "test.profile.pref", true,
+                     BraveOriginPolicyScope::kProfile);
+    return test_policies;
+  }
+
+  void InitializeManager() {
+    manager_ = BraveOriginPolicyManager::GetInstance();
+    manager_->Init(CreateBrowserTestPolicies(), CreateProfileTestPolicies(),
+                   pref_service_.get());
+  }
+
+  void SetPolicyInLocalState(const std::string& key, bool value) {
+    base::Value::Dict policies_dict =
+        pref_service_->GetDict(kBraveOriginPolicies).Clone();
+    policies_dict.Set(key, value);
+    pref_service_->SetDict(kBraveOriginPolicies, std::move(policies_dict));
+  }
+
+  base::test::TaskEnvironment task_environment_;
+  std::unique_ptr<TestingPrefServiceSimple> pref_service_;
+  raw_ptr<BraveOriginPolicyManager> manager_ = nullptr;
+};
+
+class TestObserver : public BraveOriginPolicyManager::Observer {
+ public:
+  void OnBraveOriginPoliciesReady() override { ready_called_count_++; }
+
+  int ready_called_count() const { return ready_called_count_; }
+
+ private:
+  int ready_called_count_ = 0;
```

Comment:
Perhaps this is an answer to my question in https://github.com/brave/brave-core/pull/31242/files#r2357632544. Does the count imply that the ready method can be called multiple times per session?